src/interfaces/vault.ts

```diff
@@ -330,11 +330,16 @@ export class VaultInterface<T extends ChainId> extends ServiceInterface<T> {
       if (token === EthAddress) {
         throw new SdkError("deposit:v2:eth not implemented");
       } else {
+        const shouldUsePartner = this.yearn.services.partner && this.yearn.services.partner.isAllowed(vault);
```

Comment:
I would extract this to its own function `shouldUsePartner(this.yearn.services.partner)` and use that function in line 337

Reply:
solved on 64bdcb84424cdca052f18fbae628b30e089b0076